tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp

```diff
@@ -0,0 +1,564 @@
+    gsl::not_null<std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>*>
+        functions_of_time,
+    gsl::not_null<std::mt19937*> generator, bool bad_max_radius = false,
+    bool missing_f_of_t = false) noexcept {
+  double min_radius{std::numeric_limits<double>::signaling_NaN()};
+  double max_radius{std::numeric_limits<double>::signaling_NaN()};
+  std::array<double, 3> center{};
+  generate_map_time_and_f_of_time<InteriorMap>(
+      map, time, functions_of_time, make_not_null(&min_radius),
+      make_not_null(&max_radius), make_not_null(&center), generator,
+      bad_max_radius, missing_f_of_t);
+}
+
+bool random_bool(gsl::not_null<std::mt19937*> generator) noexcept {
+  std::uniform_int_distribution<> bool_dis{0, 1};
+  return static_cast<bool>(bool_dis(*generator));
+}
+
+template <bool InteriorMap>
+void test_out_of_bounds(gsl::not_null<std::mt19937*> generator) noexcept {
+  domain::CoordinateMaps::TimeDependent::SphericalCompression<InteriorMap>
+      map{};
+  double time{std::numeric_limits<double>::signaling_NaN()};
+  std::unordered_map<std::string,
+                     std::unique_ptr<::domain::FunctionsOfTime::FunctionOfTime>>
+      functions_of_time{};
+
+  double min_radius{std::numeric_limits<double>::signaling_NaN()};
+  double max_radius{std::numeric_limits<double>::signaling_NaN()};
+  std::array<double, 3> center{};
+  generate_map_time_and_f_of_time(
+      make_not_null(&map), make_not_null(&time),
+      make_not_null(&functions_of_time), make_not_null(&min_radius),
+      make_not_null(&max_radius), make_not_null(&center), generator);
+
+  // A helper that returns a point given a radius
+  auto point = points_in_random_direction_generator(generator, center);
+
+  std::uniform_real_distribution<> rad_dis(
+      std::numeric_limits<double>::epsilon(),
+      1.0 - std::numeric_limits<double>::epsilon());
+  double radius = std::numeric_limits<double>::signaling_NaN();
+  if constexpr (InteriorMap) {
+    // Choose a radius that's out of bounds
+    radius = min_radius + rad_dis(*generator) * max_radius +
+             std::numeric_limits<double>::epsilon();
+  } else {
+    // Randomly choose on which side the radius is out of bounds
+    if (random_bool(generator)) {
+      radius = max_radius + rad_dis(*generator) * min_radius +
+               std::numeric_limits<double>::epsilon();
+    } else {
+      radius = rad_dis(*generator) * min_radius -
+               std::numeric_limits<double>::epsilon();
+    }
+  }
+  map(point(radius), time, functions_of_time);
+  std::stringstream message;
+  message << "radius should have been out of bounds, but "
+          << "radius == " << radius << ", min_radius == " << min_radius
+          << ", max_radius == " << max_radius
+          << ", InteriorMap == " << InteriorMap;
+  ERROR(message.str());
+}
+
+template <bool InteriorMap>
+void test_out_of_bounds_inverse(
+    gsl::not_null<std::mt19937*> generator) noexcept {
```

Comment:
`const gsl`

You're missing them everywhere in this file, I'll just stop commenting and let you find them with a search :)